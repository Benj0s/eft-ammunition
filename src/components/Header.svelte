<script>
  import ammunitionDataService from "../services/ammunition.data.service.js";
  import AmmoSelector from "./ammo-selector/AmmoSelector.svelte";

  async function fetchData() {
    const ammunition = await ammunitionDataService.list();

    return { ammunition };
  }
</script>

<style>
  header {
    padding: 20px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: #454545;
    color: #dedede;
  }

  .header-title {
    display: flex;
    flex-direction: column;
  }

  .header-title h2 {
    margin: 0;
    font-weight: 200;
    letter-spacing: 6px;
    line-height: 20px;
    margin-bottom: 4px;
  }

  .header-title h1 {
    margin: 0;
    font-size: 2.3rem;
    font-weight: 900;
    line-height: 32px;
  }

  @media (max-width: 610px) {
    header {
      flex-direction: column;
    }

    header :global(.ammo-selector select) {
      width: 100%;
      margin-top: 10px;
    }
  }

  @media (max-width: 390px) {
    .header-title h1 {
      font-size: 2rem;
    }
  }

  @media (max-width: 350px) {
    .header-title h1 {
      font-size: 1.6rem;
    }

    .header-title h2 {
      font-size: 1rem;
      letter-spacing: 4px;
    }
  }
</style>

<header>
  <div class="header-title">
    <h1>Ammunition Graph</h1>
    <h2>Escape from Tarkov</h2>
  </div>
  {#await fetchData() then data}
    <AmmoSelector ammunition={data.ammunition} />
  {/await}
</header>
