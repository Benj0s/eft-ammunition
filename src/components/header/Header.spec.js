import { tick } from 'svelte'
import { render, waitFor, screen } from '@testing-library/svelte'

import Header from './Header.svelte'
import ammunitionDataServiceMock from '../../services/ammunition.data.service.js'
// import ChildMock from '../../../__mocks__/childMock.svelte'

jest.mock('../../services/ammunition.data.service.js')

jest.mock('../ammo-selector/AmmoSelector.svelte', () => ({
  default: require('../../../__mocks__/childMock.svelte'),
}))

describe('Header', () => {
  it.skip('should render the component', async () => {
    ammunitionDataServiceMock.list.mockResolvedValueOnce([
      {
        id: 'test-ammo-id',
        label: 'test-ammo-label',
        class: 'test-ammo-class',
      },
    ])

    const { container } = render(Header)

    await waitFor(() => screen.getByTestId('header-actions'))
    await tick()
    // const header = screen.getByTestId('header')
    expect(container).toMatchSnapshot()
  })
})
