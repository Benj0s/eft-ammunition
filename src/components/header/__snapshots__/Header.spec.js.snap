// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<body>
  <div>
    <header
      class="svelte-146npu8"
    >
      <div
        class="header-title svelte-146npu8"
      >
        <h2
          class="svelte-146npu8"
        >
          Escape from Tarkov
        </h2>
         
        <h1
          class="svelte-146npu8"
        >
          Ammunition Graph
        </h1>
      </div>
       
      <div
        class="header-actions svelte-146npu8"
        data-testid="header-actions"
      >
        <div
          class="compare-ammo-toggle svelte-p12geb"
        >
          <div
            class="
    mdc-form-field
    
    
  "
          >
            <div
              class="mdc-switch"
            >
              <div
                class="mdc-switch__track"
              />
               
              <div
                class="mdc-switch__thumb-underlay"
              >
                <div
                  class="mdc-switch__thumb"
                >
                  <input
                    aria-checked="false"
                    class="mdc-switch__native-control "
                    id="SMUI-form-field-0"
                    role="switch"
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
             
             
            <label
              for="SMUI-form-field-0"
            >
              <span
                class="ammo-toggle-label svelte-p12geb"
                slot="label"
              >
                Compare
              </span>
            </label>
          </div>
        </div>
         
        <div
          class="ammo-selector svelte-xl2f45"
        >
          <div
            class="
    mdc-select
    
    
    mdc-select--outlined
    
    mdc-select--with-leading-icon
    
  "
          >
            <span
              slot="icon"
            >
              <i
                aria-hidden="true"
                class="mdc-select__icon material-icons"
                tabindex="-1"
              >
                <svg-mock />
              </i>
            </span>
             
            <i
              class="mdc-select__dropdown-icon"
            />
             
            <select
              class="mdc-select__native-control "
              id="SMUI-select-0"
            >
               
              <option>
                test-ammo-label
                 
              </option>
              
              
            </select>
             
             
            <div
              class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
            >
              <div
                class="mdc-notched-outline__leading"
              />
               
              <div
                class="mdc-notched-outline__notch"
                style="width: 0px;"
              >
                <label
                  class="mdc-floating-label mdc-floating-label--float-above"
                  for="SMUI-select-0"
                  style="transition-duration: 0s;"
                >
                  Ammunition
                </label>
                
                
              </div>
               
              <div
                class="mdc-notched-outline__trailing"
              />
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
